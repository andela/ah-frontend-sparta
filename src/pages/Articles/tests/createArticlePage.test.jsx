import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import { Articles } from '../index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const initialState = {
  articlesReducer:
    { error: undefined, article: undefined, articles: [] },
};
const store = mockStore(initialState);


describe('Articles Page Component', () => {
  let mountApp;
  it('Should render without crashing', () => {
    const props = {
      handleChange: jest.fn(),
      addArticle: jest.fn(),
    };
    mountApp = mount(
      <Router>
        <Provider store={store}>
          <Articles {...props} />
        </Provider>
      </Router>,
    );
    expect(mountApp).toMatchSnapshot();
  });

  it('should call an onChange function', () => {
    const props = {
      addArticle: jest.fn(),
    };
    const event = {
      preventDefault: () => {},
      target: {
        name: 'body',
        value: 'abc',
      },
    };
    const wrapper = mount(
      <Articles {...props} />,
    );
    const bodyInput = wrapper.find('#description').first();
    bodyInput.simulate('change', event);
    expect(event.target.value).toEqual('abc');
  });

  it('should call a submit a form', () => {
    const props = {
      addArticle: jest.fn(),
    };
    const wrapper = mount(
      <Articles {...props} />,
    );
    const instance = wrapper.instance();
    instance.onSubmit({ preventDefault: jest.fn() });
    expect(props.addArticle).toHaveBeenCalled();
  });
});
