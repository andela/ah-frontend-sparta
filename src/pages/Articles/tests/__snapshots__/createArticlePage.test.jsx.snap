// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles Page Component Should render without crashing 1`] = `ReactWrapper {}`;
